From 3436b513e3872cbe71fdea3d09f7e89d1d85612e Mon Sep 17 00:00:00 2001
From: Markus Armbruster <armbru@redhat.com>
Date: Fri, 25 Nov 2011 17:07:01 +0100
Subject: [PATCH 08/11] qdev: Fix crash on -device '?=x'

RH-Author: Markus Armbruster <armbru@redhat.com>
Message-id: <1322240821-31350-1-git-send-email-armbru@redhat.com>
Patchwork-id: 35450
O-Subject: [RHEL-6.3 PATCH qemu-kvm] qdev: Fix crash on -device '?=x'
Bugzilla: 757142
RH-Acked-by: Jeffrey Cody <jcody@redhat.com>
RH-Acked-by: Paul Moore <pmoore@redhat.com>
RH-Acked-by: Kevin Wolf <kwolf@redhat.com>

Spotted by Coverity.

Signed-off-by: Markus Armbruster <armbru@redhat.com>
Signed-off-by: Stefan Hajnoczi <stefanha@linux.vnet.ibm.com>
(cherry picked from commit 542379f42642179cea33571c2799ca6fd03bf452)
---
Bug 757142
brew: http://brewweb.devel.redhat.com/brew/taskinfo?taskID=3852167

 hw/qdev.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

Signed-off-by: Michal Novotny <minovotn@redhat.com>
---
 hw/qdev.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/hw/qdev.c b/hw/qdev.c
index 453c17f..eafb463 100644
--- a/hw/qdev.c
+++ b/hw/qdev.c
@@ -170,7 +170,7 @@ int qdev_device_help(QemuOpts *opts)
         return 1;
     }
 
-    if (!qemu_opt_get(opts, "?")) {
+    if (!driver || !qemu_opt_get(opts, "?")) {
         return 0;
     }
 
-- 
1.7.7.3

